### PR TITLE
Fix incorrect file permissions in Dangerzone RPMs

### LINKED ISSUE #727: Wrong permissions on built Fedora artifacts.
While testing the produced RPMs for the 0.6.0 release, we stumbled upon this issue on Fedora 39 (Fedora 38 is also affected):

image

It seems that the produced RPMs have wrong permissions (current: `rw-------` / `600`, expected: `rw-r--r--` / `644`), specifically on the following files:

```
dangerzone/conversion/common.py
dangerzone/conversion/doc_to_pixels.py
dangerzone/conversion/pixels_to_pdf.py
qubes/dz.Convert
qubes/dz.ConvertDev
```

While rebuilding the RPMs, we see the following warnings in the logs:

```
*** WARNING: ./usr/lib/python3.12/site-packages/dangerzone-0.6.0.dist-info/LICENSE is executable but has no shebang, removing executable bit
*** WARNING: ./usr/lib/python3.12/site-packages/dangerzone-0.6.0.dist-info/WHEEL is executable but has no shebang, removing executable bit
*** WARNING: ./usr/lib/python3.12/site-packages/dangerzone-0.6.0.dist-info/entry_points.txt is executable but has no shebang, removing executable bit
*** WARNING: ./usr/lib/python3.12/site-packages/dangerzone-0.6.0.dist-info/INSTALLER is executable but has no shebang, removing executable bit
*** WARNING: ./usr/lib/python3.12/site-packages/dangerzone-0.6.0.dist-info/METADATA is executable but has no shebang, removing executable bit
*** WARNING: ./usr/lib/python3.12/site-packages/dangerzone/util.py is executable but has no shebang, removing executable bit
*** WARNING: ./usr/lib/python3.12/site-packages/dangerzone/__init__.py is executable but has no shebang, removing executable bit
*** WARNING: ./usr/lib/python3.12/site-packages/dangerzone/logic.py is executable but has no shebang, removing executable bit
*** WARNING: ./usr/lib/python3.12/site-packages/dangerzone/container-pip-requirements.txt is executable but has no shebang, removing executable bit
*** WARNING: ./usr/lib/python3.12/site-packages/dangerzone/cli.py is executable but has no shebang, removing executable bit
*** WARNING: ./usr/lib/python3.12/site-packages/dangerzone/document.py is executable but has no shebang, removing executable bit
*** WARNING: ./usr/lib/python3.12/site-packages/dangerzone/gui/main_window.py is executable but has no shebang, removing executable bit
*** WARNING: ./usr/lib/python3.12/site-packages/dangerzone/gui/__init__.py is executable but has no shebang, removing executable bit
*** WARNING: ./usr/lib/python3.12/site-packages/dangerzone/gui/updater.py is executable but has no shebang, removing executable bit
*** WARNING: ./usr/lib/python3.12/site-packages/dangerzone/gui/logic.py is executable but has no shebang, removing executable bit
mangling shebang in /usr/lib/python3.12/site-packages/dangerzone/conversion/doc_to_pixels.py from /usr/bin/env python3 to #!/usr/bin/python3
mangling shebang in /usr/lib/python3.12/site-packages/dangerzone/conversion/common.py from /usr/bin/env python3 to #!/usr/bin/python3
mangling shebang in /usr/lib/python3.12/site-packages/dangerzone/conversion/pixels_to_pdf.py from /usr/bin/env python3 to #!/usr/bin/python3
*** WARNING: ./usr/lib/python3.12/site-packages/dangerzone/conversion/errors.py is executable but has no shebang, removing executable bit
*** WARNING: ./usr/lib/python3.12/site-packages/dangerzone/settings.py is executable but has no shebang, removing executable bit
*** WARNING: ./usr/lib/python3.12/site-packages/dangerzone/args.py is executable but has no shebang, removing executable bit
*** WARNING: ./usr/lib/python3.12/site-packages/dangerzone/errors.py is executable but has no shebang, removing executable bit
*** WARNING: ./usr/lib/python3.12/site-packages/dangerzone/isolation_provider/qubes.py is executable but has no shebang, removing executable bit
*** WARNING: ./usr/lib/python3.12/site-packages/dangerzone/isolation_provider/container.py is executable but has no shebang, removing executable bit
*** WARNING: ./usr/lib/python3.12/site-packages/dangerzone/isolation_provider/dummy.py is executable but has no shebang, removing executable bit
*** WARNING: ./usr/lib/python3.12/site-packages/dangerzone/isolation_provider/base.py is executable but has no shebang, removing executable bit
*** WARNING: ./usr/share/applications/press.freedom.dangerzone.desktop is executable but has no shebang, removing executable bit
*** WARNING: ./usr/share/dangerzone/image-id.txt is executable but has no shebang, removing executable bit
*** WARNING: ./usr/share/dangerzone/dangerzone.css is executable but has no shebang, removing executable bit
*** WARNING: ./usr/share/dangerzone/version.txt is executable but has no shebang, removing executable bit
mangling shebang in /etc/qubes-rpc/dz.ConvertDev from /usr/bin/env python3 to #!/usr/bin/python3
mangling shebang in /etc/qubes-rpc/dz.Convert from /bin/sh to #!/usr/bin/sh
```

We can see here that the files with the wrong permissions are the ones whose shebang has been mangled. The Fedora docs further explain this operation: https://docs.fedoraproject.org/en-US/packaging-guidelines/#_shebang_lines

What's interesting though is that this warning is not present in the CI job that builds RPMs, nor is it reproduced locally: https://github.com/freedomofpress/dangerzone/actions/runs/8041707585/job/21961324246#step:5:207.

### PR DESCRIPTION
**Pull request by apyrgio**
Switch build directory for the `rpmbuild` command from `./install/linux/rpm-build` to `~/rpmbuild`. The main reason for this is that we want a build directory that will not be mounted in the container, since we've experienced issues with file permissions.

Regarding the choice of directories, we went with `~/rpmbuild` because it's outside the Dangerzone source, and also because it's the default choice in Fedora [1].

In addition to the above fix, make the following changes:
1. Add a check to detect if a Dangezone RPM is about to be built with wrong permissions.
2. Fix some shebangs that were the cause of complaints during RPM build.

[1]: https://github.com/rsrchboy/rpmdevtools/blob/3ae1eeafeeda19e7bcb4546300fe551fd879e0ca/rpmdev-setuptree#L60
    
Closes #727

### CODE REVIEW

File: install/linux/build-rpm.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
 #!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 import argparse
+import getpass
```
Comment (almet) [inline install/linux/build-rpm.py:4]: Why is this import required? 🤔

File: install/linux/dangerzone.spec [hunk 2]
```diff
@@ -258,6 +258,17 @@ install -m 755 -d %{buildroot}/etc/qubes-rpc
...
 %endif
 
+%check
+# Detect if the filesystem has been affecting our file permissions.
+bad_files=$(find %{buildroot} -perm 0600)
+if [ -n "${bad_files}" ]; then
+    echo "Error while building the Dangerzone RPM. Detected the following files with wrong permissions (600):"
+    echo ${bad_files}
+    echo ""
+    echo "For more info about this error, see https://github.com/freedomofpress/dangerzone/issues/727"
```
Comment (almet) [inline install/linux/dangerzone.spec:268]: I really find these messages useful, especially as they point to the full context with the github issues. 🤌

File: install/linux/build-rpm.py [hunk 3]
```diff
@@ -1,6 +1,7 @@
...
 import inspect
 import os
 import pathlib
```
Comment (almet) [inline install/linux/build-rpm.py:7]: nitpick: As we're only using `pathlib.Path` in this file, we might just import it:

```python
from pathlib import Path
```

**Review by almet — APPROVED** [review]
Congrats for you work on this :-) I've let a few comments, but It looks good to me in general 👍

File: install/linux/build-rpm.py (hunk 1 above)
Comment (apyrgio) [reply install/linux/build-rpm.py]: Whoops, that's no longer required, Removed it in bdf0204dd33730fc294c82bcd934a35f940da6e2.

File: install/linux/build-rpm.py (hunk 3 above)
Comment (apyrgio) [reply install/linux/build-rpm.py]: You're right, fixed in bdf0204dd33730fc294c82bcd934a35f940da6e2.

File: install/linux/build-rpm.py [hunk 4]
```diff
@@ -72,7 +73,7 @@ def build(qubes=False):
         container_tar_gz.rename(container_tar_gz_bak)
     try:
         subprocess.run(["poetry", "build", "-f", "sdist"], cwd=root, check=True)
-        os.rename(dist_path / sdist_name, build_dir / "SOURCES" / sdist_name)
+        shutil.copy2(dist_path / sdist_name, build_dir / "SOURCES" / sdist_name)
```
[the remark this answers is not in the document]
Comment (apyrgio) [reply install/linux/build-rpm.py]: We have switched here because `os.rename` does not work across filesystem boundaries. Unlike Linux's `mv` command (which copies the file and then unlinks it, in the case of different filesystems), `os.rename`  simply fails to move the file. So, for compatibility reasons, a copy-unlink approach is better.

I have add a comment in bdf0204dd33730fc294c82bcd934a35f940da6e2 to clarify this (hopefully).
